example/lib/main.dart

```diff
@@ -80,6 +80,11 @@ class _MyHomePageState extends State<MyHomePage> {
                   Map data,
                   String error,
                 }) {
+                  if (data.isNotEmpty) {
+                    repository['viewerHasStarred'] =
+                      data['addStar']['starrable']['viewerHasStarred'];
+                  }
```

Comment:
Until we properly restructure the result data, this is a good alternative for the optimistic UI updates.